internal/apidiff/apidiff.go

```diff
@@ -0,0 +1,255 @@
+type manifestEntry struct {
+	DistributionName  string `json:"distribution_name"`
+	Description       string `json:"description"`
+	Language          string `json:"language"`
+	ClientLibraryType string `json:"client_library_type"`
+	DocsURL           string `json:"docs_url"`
+	ReleaseLevel      string `json:"release_level"`
+}
+
+func diffModules() {
+	goCloud := cloneGoCloud()
+	defer os.RemoveAll(goCloud)
+
+	for imp, entry := range manifest {
+		// Only diff stable clients.
+		if entry.ReleaseLevel != "ga" {
+			continue
+		}
+
+		// Prepare module directory paths relative to the repo root.
+		pkg := strings.TrimPrefix(imp, rootMod+"/")
+		baseModDir := goCloud
+		modDir := root
+
+		// Manual clients are also submodules, so we need to run apidiff in the
+		// submodule.
+		if entry.ClientLibraryType == "manual" {
+			baseModDir = path.Join(baseModDir, pkg)
+			modDir = path.Join(modDir, pkg)
+		}
+
+		// Create apidiff base from repo remote HEAD.
+		base, err := writeBase(baseModDir, imp, pkg)
+		if err != nil {
+			errs[imp] = err
+			continue
+		}
+
+		// Diff the current checked out change against remote HEAD base.
+		out, err := diff(modDir, imp, pkg, base)
+		if err != nil {
+			errs[imp] = err
+			continue
+		}
+
+		if out != "" {
+			diffs[imp] = out
+		}
+	}
+}
+
+func writeBase(baseModDir, imp, pkg string) (string, error) {
+	if err := cd(baseModDir); err != nil {
+		return "", err
+	}
+
+	base := path.Join(baseModDir, "pkg.master")
+	out, err := execE("apidiff", "-w", base, imp)
+	if err != nil && !isSubModErr(out) {
+		return "", err
+	}
+
+	// If there was an issue with loading a submodule, change into that
+	// submodule directory and try again.
+	if isSubModErr(out) {
+		parent := manualParent(imp, manifest)
+		if parent == pkg {
+			return "", fmt.Errorf("unable to find parent module for %q", imp)
+		}
+		if err := cd(parent); err != nil {
+			return "", err
+		}
+		out, err := execE("apidiff", "-w", base, imp)
+		if err != nil {
+			return "", fmt.Errorf("%s: %s", err, out)
+		}
+	}
+	return base, nil
+}
+
+func diff(modDir, imp, pkg, base string) (string, error) {
+	if err := cd(modDir); err != nil {
+		return "", err
+	}
+	out, err := execE("apidiff", "-incompatible", base, imp)
+	if err != nil && !isSubModErr(out) {
+		return "", err
+	}
+	if isSubModErr(out) {
+		parent := manualParent(imp, manifest)
+		if parent == pkg {
+			return "", fmt.Errorf("unable to find parent module for %q", imp)
+		}
+		if err := cd(parent); err != nil {
+			return "", err
+		}
+		out, err = execE("apidiff", "-w", base, imp)
+		if err != nil {
+			return "", fmt.Errorf("%s: %s", err, out)
+		}
+	}
+
+	return out, err
+}
+
+func shouldRun() {
+	head := mustExec("git", "log", "-1")
+	if strings.Contains(head, "BREAKING_CHANGE_ACCEPTABLE") {
+		os.Exit(0)
+	}
+}
+
+func loadManifest() {
+	f, err := os.Open(repoMetadataPath)
+	if err != nil {
+		log.Fatalln(err)
+	}
+	err = json.NewDecoder(f).Decode(&manifest)
```

Comment:
nit: Can be combined with the line below.

Reply:
Done.